cmd/dra-example-kubeletplugin/health.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2025 The Kubernetes Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"sync"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+	drapb "k8s.io/kubelet/pkg/apis/dra/v1beta1"
+	registerapi "k8s.io/kubelet/pkg/apis/pluginregistration/v1"
+
+	"sigs.k8s.io/dra-example-driver/pkg/consts"
+)
+
+type healthcheck struct {
+	grpc_health_v1.UnimplementedHealthServer
+
+	server *grpc.Server
+	wg     sync.WaitGroup
+
+	regClient registerapi.RegistrationClient
+	draClient drapb.DRAPluginClient
+}
+
+func startHealthcheck(ctx context.Context, config *Config) (*healthcheck, error) {
+	log := klog.FromContext(ctx)
+
+	port := config.flags.healthcheckPort
+	if port == 0 {
+		return nil, nil
+	}
+
+	addr := net.JoinHostPort("", strconv.Itoa(port))
+	lis, err := net.Listen("tcp", addr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to listen for healthcheck service at %s: %w", addr, err)
+	}
+
+	// TODO: this needs to adapt when seamless upgrades
+	// are enabled and the filename includes a uid.
+	regSockPath, err := url.JoinPath("unix:///", config.flags.kubeletRegistrarDirectoryPath, consts.DriverName+"-reg.sock")
```

Comment:
Should it be "unix://" ? Third slash is a part of an absolute path and `config.flags.kubeletRegistrarDirectoryPath` **is** an absolute path, right?
Alternatively you can use `url.URL{Scheme: "unix", Path: ... }.String()`.

Reply:
`url.JoinPath` swallows the leading slash on the first path element beyond the base, so this was kind of an ugly hack to compensate for that: https://go.dev/play/p/Hrrzu0qCQu0
```go
func main() {
	u, _ := url.JoinPath("unix://", "/var")
	fmt.Println(u)
}
```
```
unix://var
```

I agree that constructing a literal `url.URL` looks better than this so I've updated to do that.